packages/compass-preferences-model/index.d.ts

```diff
@@ -1,15 +1,31 @@
-declare class CompassPreferencesModel {
-  fetch: () => void;
-  save: () => void;
-  set: (
-    key: string,
-    value: unknown
-  ) => void;
-};
-export default CompassPreferencesModel;
+import { preferencesIpc } from '.';
 
 export enum THEMES {
   DARK = 'DARK',
   LIGHT = 'LIGHT',
   OS_THEME = 'OS_THEME'
-};
+}
+
+export type UserPreferences = {
+  showedNetworkOptIn: boolean; // Has the settings dialog has been shown before.
+  autoUpdates: boolean;
+  enableMaps: boolean;
+  trackErrors: boolean;
+  trackUsageStatistics: boolean;
+  enableFeedbackPanel: boolean;
+  theme: THEMES;
+}
+
+export type GlobalPreferences = UserPreferences; // TODO: extend with global preferences.
+
+declare class Preferences {
+  constructor(basepath?: string): void;
+  fetchPreferences(): Promise<GlobalPreferences>;
+  savePreferences(attributes: GlobalPreferences): Promise<GlobalPreferences>;
+  getPreferences(): Promise<GlobalPreferences>;
+  getConfigurableUserPreferences(): Promise<UserPreferences>;
```

Comment:
Just a note here, but it’s surprising that this is named `get...` but sets defaults, and we’re not going to be able to keep this signature in the future (because there will be preferences that are fixed through the CLI or the global config and `Promise<UserPreferences>` does not communicate that). We can take care of that when we get to it, though.